pay-api/src/pay_api/services/eft_short_name_historical.py

```diff
@@ -0,0 +1,194 @@
+    limit: Optional[int] = 10
+
+
+class EFTShortnameHistorical:
+    """Service to manage EFT Short name historical data."""
+
+    @staticmethod
+    def create_funds_received(history: EFTShortnameHistory) -> EFTShortnamesHistoricalModel:
+        """Create EFT Short name funds received historical record."""
+        return EFTShortnamesHistoricalModel(
+            amount=history.amount,
+            created_by='SYSTEM',
+            credit_balance=history.credit_balance,
+            description='Funds Received',
+            hidden=history.hidden,
+            is_processing=history.is_processing,
+            short_name_id=history.short_name_id,
+            transaction_date=EFTShortnameHistorical.transaction_date_now(),
+            transaction_type=EFTHistoricalTypes.FUNDS_RECEIVED.value
+        )
+
+    @staticmethod
+    @user_context
+    def create_statement_paid(history: EFTShortnameHistory, **kwargs) -> EFTShortnamesHistoricalModel:
+        """Create EFT Short name statement paid historical record."""
+        return EFTShortnamesHistoricalModel(
+            amount=history.amount,
+            created_by=kwargs['user'].user_name,
+            credit_balance=history.credit_balance,
+            description='Statement Paid',
+            hidden=history.hidden,
+            is_processing=history.is_processing,
+            payment_account_id=history.payment_account_id,
+            related_group_link_id=history.related_group_link_id,
+            short_name_id=history.short_name_id,
+            statement_number=history.statement_number,
+            transaction_date=EFTShortnameHistorical.transaction_date_now(),
+            transaction_type=EFTHistoricalTypes.STATEMENT_PAID.value
+        )
+
+    @staticmethod
+    @user_context
+    def create_statement_reverse(history: EFTShortnameHistory, **kwargs) -> EFTShortnamesHistoricalModel:
+        """Create EFT Short name statement reverse historical record."""
+        return EFTShortnamesHistoricalModel(
+            amount=history.amount,
+            created_by=kwargs['user'].user_name,
+            credit_balance=history.credit_balance,
+            description='Payment Reversed',
+            hidden=history.hidden,
+            is_processing=history.is_processing,
+            payment_account_id=history.payment_account_id,
+            related_group_link_id=history.related_group_link_id,
+            short_name_id=history.short_name_id,
+            statement_number=history.statement_number,
+            transaction_date=EFTShortnameHistorical.transaction_date_now(),
+            transaction_type=EFTHistoricalTypes.STATEMENT_REVERSE.value
+        )
+
+    @staticmethod
+    def transaction_date_now() -> datetime:
+        """Construct a datetime using the legislation timezone."""
+        return datetime.now(tz=timezone.utc)
+
+    @staticmethod
+    def _get_account_name():
+        """Return case statement for deriving payment account name."""
+        return case(
```

Comment:
Didn't we have this code somewhere else?

Reply:
yea it used to be in the transaction service, which this service replaces. I kept this piece.